packages/web3-utils/src/numberToBN.js

```diff
@@ -0,0 +1,67 @@
+// This file was internalized from silentcicero/number-to-bn
+// in order to reduce the bundle size. Package shared a
+// bn.js dependency which wasn't being updated resulting
+// in duplicate versions of that dep included in webpacks
+
+/*
+The MIT License
+
+Copyright (c) 2016 Nick Dodson. nickdodson.com
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+var BN = require('bn.js');
+var stripHexPrefix = require('strip-hex-prefix');
+
+/**
+ * Returns a BN object, converts a number value to a BN
+ * @param {String|Number|Object} `arg` input a string number, hex string number, number, BigNumber or BN object
+ * @return {Object} `output` BN object of the number
+ * @throws if the argument is not an array, object that isn't a bignumber, not a string number or number
+ */
+module.exports = function numberToBN(arg) {
```

Comment:
Is this function exposed in web3's API? If not, I'd remove it. It's a a huge mess.

Reply:
This will be used internally for some utility functions but could probably get replaced with the ethers BigNumber wrapper object.

Reply:
It internal to [toBN][1] and [fromWei][2] both of which are heavily used.  

Personally I'd be more comfortable removing this in a different PR that rationalized use of Ethers utils and where we have tests in place to make sure we don't accidentally introduce unexpected behavior. Also where we double-check the Ethers issues about differences people see in how Ethers utils behave vs. Web3 since the code here is 3 years old. 

PR bumps BN across some patch versions - might be worth isolating.

[1]: https://github.com/ethereum/web3.js/blob/6ca8bba4e5cb9f76672b326541ba6c620af40e70/packages/web3-utils/src/utils.js#L61-L67
[2]: https://github.com/ethereum/web3.js/blob/aeb1f467923671f933d5ab8a4bbd688d6a792bf9/packages/web3-utils/src/ethjsUnit.js#L111